mam/v2/prng/tests/test_prng_types.c

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright (c) 2018 IOTA Stiftung
+ * https:github.com/iotaledger/entangled
+ *
+ * MAM is based on an original implementation & specification by apmi.bsu.by
+ * [ITSec Lab]
+ *
+ * Refer to the LICENSE file for licensing information
+ */
+
+#include <string.h>
+
+#include <unity/unity.h>
+
+#include "common/trinary/trit_tryte.h"
+#include "mam/v2/prng/mam_prng_t_set.h"
+#include "mam/v2/prng/prng.h"
+#include "mam/v2/prng/prng_types.h"
+
+static bool mam_prng_t_set_cmp(mam_prng_t_set_t const set1,
+                               mam_prng_t_set_t const set2) {
+  mam_prng_t_set_entry_t *entry1 = NULL;
+  mam_prng_t_set_entry_t *tmp1 = NULL;
+
+  mam_prng_t_set_entry_t *entry2 = NULL;
+  mam_prng_t_set_entry_t *tmp2 = NULL;
+
+  size_t equal_elements_count = 0;
+
+  TEST_ASSERT_EQUAL_INT(mam_prng_t_set_size(set1), mam_prng_t_set_size(set2));
+
+  // We have to loop in O(N^2) because mam_ntru_sk_t_set_contains compares hash
+  // value based on
+  // the address assigned into the "f" field in the ntru_sk_t type
+  HASH_ITER(hh, set1, entry1, tmp1) {
+    HASH_ITER(hh, set2, entry2, tmp2) {
+      if (!memcmp(&entry1->value.secret_key, &entry2->value.secret_key,
+                  mam_prng_serialized_size())) {
+        ++equal_elements_count;
+      }
+    }
+  }
+
+  TEST_ASSERT_EQUAL_INT(mam_prng_t_set_size(set1), equal_elements_count);
```

Comment:
nits: you can just return the comparison of both since the result will be tested